File: history_today-master/app/src/main/java/com/animee/todayhistory/MainContract.java
package com.animee.todayhistory;

import com.animee.todayhistory.bean.HistoryBean;
import com.animee.todayhistory.bean.HistoryDescBean;
import com.animee.todayhistory.bean.LaoHuangliBean;
import com.animee.todayhistory.bean.LaoHuangliHoursBean;

public interface MainContract {
    interface MainView {
        void getHuangLiSuccess(LaoHuangliBean bean);

        void getHuangLiError(Throwable throwable);
        
        void getHuangLiHoursSuccess(LaoHuangliHoursBean bean);
        
        void getHuangLiHoursError(Throwable throwable);

        void getHistorySuccess(HistoryBean bean);

        void getHistoryError(Throwable throwable);

    }
    interface Presenter{
        void loadGet(String his);
        void getHuangLi();
        void getHuangLiHours();
        void getHistory();
        void UpdateTime(int year, int month, int dayOfMonth);
    }
}

File: history_today-master/app/src/main/java/com/animee/todayhistory/MainContract.kt
package com.animee.todayhistory

import com.animee.todayhistory.bean.HistoryBean
import com.animee.todayhistory.bean.LaoHuangliBean
import com.animee.todayhistory.bean.LaoHuangliHoursBean

interface MainContract {
    interface MainView {
        fun getHuangLiSuccess(bean: LaoHuangliBean)
        fun getHuangLiError(throwable: Throwable)
        fun getHuangLiHoursSuccess(bean: LaoHuangliHoursBean)
        fun getHuangLiHoursError(throwable: Throwable)
        fun getHistorySuccess(bean: HistoryBean)
        fun getHistoryError(throwable: Throwable)
    }

    interface Presenter {
        fun loadGet(his: String)
        fun getHuangLi()
        fun getHuangLiHours()
        fun getHistory()
        fun UpdateTime(year: Int, month: Int, dayOfMonth: Int)
    }
}
